### PR TITLE
roachprod: support customized builds

### PR DESCRIPTION
**Pull request by rail**
This PR adds support for customized build location that can be used for staging a roachprod cluster.

Epic: none
Release note: None

### CODE REVIEW

File: pkg/cmd/roachprod/main.go [hunk 1]
```diff
@@ -860,11 +860,13 @@ var stageURLCmd = &cobra.Command{
...
-              latest build version is used.
-  workload  - Cockroach workload application.
-  release   - Official CockroachDB Release. Must provide a specific release
-              version.
+  cockroach  - Cockroach Unofficial. Can provide an optional SHA, otherwise
+               latest build version is used.
+  workload   - Cockroach workload application.
+  release    - Official CockroachDB Release. Must provide a specific release
+               version.
+  customized - Customized CockroachDB build. Must provide a specific build
```
Comment (srosenberg) [inline pkg/cmd/roachprod/main.go:868]: How is customized build different from "Cockroach Unofficial"? It's not immediately clear from the description.

Comment (rail) [reply pkg/cmd/roachprod/main.go:868]: Good question. Cockroach Unofficial is built using merged source code (aka edge binaries). Customized builds are built using using `./scripts/tag-custom-build.sh` and the source code is not merged to the main repo (usually). They live in different buckets as well.

Comment (rail) [reply pkg/cmd/roachprod/main.go:868]: I can update the comments for both types.

**Review by srosenberg — APPROVED** [review]

**Comment by rail** [comment]
bors r=srosenberg

**Pull request merged**